components/match2/wikis/smite/match_group_input_custom.lua

```diff
@@ -0,0 +1,517 @@
+		match.finished = true
+	end
+
+	-- see if match should actually be finished if score is set
+	if not Logic.readBool(match.finished) then
+		matchFunctions._finishMatch(match, opponents, isScoreSet)
+	end
+
+	-- apply placements and winner if finshed
+	if Logic.readBool(match.finished) then
+		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+---@param match table
+---@return boolean
+function matchFunctions._finishMatch(match, opponents, isScoreSet)
+	-- If a winner has been set
+	if Logic.isNotEmpty(match.winner) then
+		match.finished = true
+	end
+
+	-- If special status has been applied to a team
+	if CustomMatchGroupInput.placementCheckSpecialStatus(opponents) then
+		match.finished = true
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	match.finished = Logic.readBool(match.finished)
+		or isScoreSet and (
+			Array.any(opponents, function(opponent) return (tonumber(opponent.score) or 0) > match.bestof/2 end)
+			or Array.all(opponents, function(opponent) return (tonumber(opponent.score) or 0) == match.bestof/2 end)
+		)
+
+	-- If enough time has passed since match started, it should be marked as finished
+	if isScoreSet and match.timestamp ~= DateExt.defaultTimestamp then
+		local threshold = match.dateexact and SECONDS_UNTIL_FINISHED_EXACT
+			or SECONDS_UNTIL_FINISHED_NOT_EXACT
+		if match.timestamp + threshold < NOW then
+			match.finished = true
+		end
+	end
+
+	return match
+end
+
+function matchFunctions._makeAllOpponentsLoseByWalkover(opponents, walkoverType)
+	for index in pairs(opponents) do
+		opponents[index].score = NOT_PLAYED_SCORE
+		opponents[index].status = walkoverType
+	end
+	return opponents
+end
+
+--
+-- map related functions
+--
+
+-- Parse extradata information
+---@param map table
+---@return table
+function mapFunctions.getAdditionalExtraData(map)
+	map.extradata.comment = map.comment
+	map.extradata.team1side = string.lower(map.team1side or '')
+	map.extradata.team2side = string.lower(map.team2side or '')
+
+	return map
+end
+
+-- Parse participant information
+---@param map table
+---@return table
+function mapFunctions.getParticipants(map, opponents)
+	local participants = {}
+	local heroData = {}
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		for playerIndex = 1, MAX_NUM_PLAYERS do
+			local hero = map['t' .. opponentIndex .. 'g' .. playerIndex]
+			heroData['team' .. opponentIndex .. 'hero' .. playerIndex] = HeroNames[hero]
+		end
+
+		for _, ban, banIndex in Table.iter.pairsByPrefix(map, 't' .. opponentIndex .. 'b') do
+			heroData['team' .. opponentIndex .. 'ban' .. banIndex] = HeroNames[ban]
+		end
+	end
+
+	map.extradata = heroData
+	map.participants = participants
+	return mapFunctions.getAdditionalExtraData(map)
+end
```

Comment:
```suggestion
	local heroData = {}
	for opponentIndex = 1, MAX_NUM_OPPONENTS do
		for playerIndex = 1, MAX_NUM_PLAYERS do
			local hero = map['t' .. opponentIndex .. 'g' .. playerIndex]
			heroData['team' .. opponentIndex .. 'hero' .. playerIndex] = HeroNames[hero and hero:lower()]
			
			local ban = map['t' .. opponentIndex .. 'b' .. playerIndex]
			heroData['team' .. opponentIndex .. 'ban' .. banIndex] = HeroNames[ban and ban:lower()]
		end
	end

	map.extradata = heroData
	return map
end
```

Each team picks 5 gods and bans other 5 so `Table.iter.pairsByPrefix`  feels unnecessary here imo.